### ISSUE TITLE
Commit subject length incorrectly detected as `> 90` chars

### ISSUE THREAD

**Issue by polarathene** [vbatts/git-validation#61]
I experienced this bug [here](https://github.com/containerd/containerd/pull/8924#issuecomment-1719651375)
- Original commit failing: https://github.com/containerd/containerd/commit/364bf956a0aa575ffb3d66dd56b9ae04775ab224
- Two blank lines to separate subject passes: https://github.com/containerd/containerd/commit/3ca39ef01608fdd44245c0173bf071682b3bfe3c

Failure occurring at:

https://github.com/vbatts/git-validation/blob/be39cec199fb8a02191a1a100ca12c70e059586b/rules/shortsubject/shortsubject.go#L32-L37

https://github.com/vbatts/git-validation/blob/be39cec199fb8a02191a1a100ca12c70e059586b/git/commits.go#L102

Presumably related to what's going on here (_but I've not troubleshooted further_):

https://github.com/vbatts/git-validation/blob/be39cec199fb8a02191a1a100ca12c70e059586b/git/commits.go#L177-L184

---

Failing commit message:

```markdown
fix: Remove `LimitNOFILE` from `containerd.service`

Remove `LimitNOFILE` from `containerd.service` to rely on the systemd v240 implicit default of `1024:524288`. On supported platforms with systemd prior to v240, packagers will patch the service with an explicit `LimitNOFILE=1024:524288`.

- `1024` soft limit is an implicit default, avoiding unexpected breakage. Software that needs a higher limit should request to raise the soft limit for its process.
- `524288` hard limit is an implicit default since systemd v240 and is adequate for most processes (_half of the historical limit from `fs.nr_open` of `1048576`_), while 4096 is the implicit default from the kernel (often too low).
- The hard limit may not exceed `fs.nr_open` (_which a value of `infinity` will resolve to_). On most systems with systemd v240 or newer, this will resolve to an excessive size of 2^30 (over 1 billion).
- When set to `infinity` (usually as the soft limit) software may experience significantly increased resource usage, resulting in a performance regression or runtime failures that are difficult to troubleshoot.

Signed-off-by: Brennan Kinney <5098581+polarathene@users.noreply.github.com>
```

**FAIL** - [Results in](https://github.com/containerd/containerd/actions/runs/5779731320/job/15662426033#step:4:235):

```
 * 364bf956 "fix: Remove `LimitNOFILE` from `containerd.service` Remove `LimitNOFILE` from `containerd.service` to rely on the systemd v240 implicit default of `1024:524288`. On supported platforms with systemd prior to v240, packagers will patch the service with an explicit `LimitNOFILE=1024:524288`." ... FAIL
    - PASS - commit does not have any whitespace errors
    - PASS - has a valid DCO
    - FAIL - commit subject exceeds 90 characters
```

**PASS** - While [adding another blank line after the subject was successful](https://github.com/containerd/containerd/actions/runs/6190868080/job/16808014636?pr=8924#step:4:251) 🤷‍♂️ 

```
   * 3ca39ef0 "fix: Remove `LimitNOFILE` from `containerd.service`" ... PASS
    - PASS - commit does not have any whitespace errors
    - PASS - has a valid DCO
    - PASS - commit subject is 72 characters or less! *yay*
```

Not sure if it's due to the content/formatting potentially throwing it off? Both locally on my git client and Github's own web view of the commit, presented the correct commit subject with the original commit (_21 chars to spare within the 72 limit_).

**Comment by vbatts** [comment]
I'm trying to troubleshoot and really unsure what's happening here. Like, why it pulled-in the next lines as the "subject" line... Like, I wonder if it was some funky CRLF ?

I couldn't fetch https://github.com/containerd/containerd/commit/364bf956a0aa575ffb3d66dd56b9ae04775ab224 to a local git client. Seems it's only viewable on github.
